analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,859 @@
+        }
+
+Example of usage
+
+.. code-block:: html
+
+    <button onClick = function(){_paq.push(["trackLink", "http://www.example.com/example.xrt", "download"]);}>
+        Click me!
+    </button>
+
+
+After each outbound link there is small time frame after which the file will download that will
+ensure there is enough time to track download.
+That time frame is set to 500ms by default. To modify it you can use ``setLinkTrackingTimer`` function::
+
+    _paq.push(["setLinkTrackingTimer" time]);
+
+.. data:: time
+
+    **Required** Time in ms between user interaction and downloading file.
+
+
+Disabling tracking
+``````````````````
+
+You can disable download and outlink tracking for links using CSS classes, simply add ``piwik_ignore`` css class.
+
+To disable using CSS class you can use ``setIgnoreClassess`` function::
+
+    _paq.push(["setIgnoreClasses", className);
+
+.. data:: className
+
+    **Required** Css class name that will be ignored
+
+
+User ID Management
+^^^^^^^^^^^^^^^^^^
+User ID enables merging user data that is collected between many devices and browsers.
+
+You must provide unique user-id for every user. To set user ID for tracked data use ``setUserId`` function::
+
+    _paq.push(["setUserId", userID]);
+
+.. data:: userID
+
+    **Required** Unique, non empty string preserved for each user.
+
+Tracking domains and subdomains
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tracking single domain
+``````````````````````
+To track single domain name without tracking subdomains (or single subdomain) use default snippet code::
+
+    _paq.push(["setSiteId", 1]);
+    _paq.push(["setTrackerUrl", u+"piwik.php"]);
+    _paq.push(["trackPageView"]);
+
+Tracking domains and all subdomains of the website
+``````````````````````````````````````````````````
+To track all data between domain and all its subdomains we must use cookies using this snippet::
+
+    _paq.push(["setSiteId", 1]);
+    _paq.push(["setTrackerUrl", u+"piwik.php"]);
+
+    // Share the tracking cookie across example.com, www.example.com, subdomain.example.com, ...
+    _paq.push(["setCookieDomain", "*.example.com"]);
+
+    // Tell Piwik the website domain so that clicks on these domains are not tracked as "Outlinks"
+    _paq.push(["setDomains", "*.example.com"]);
+
+    _paq.push(["trackPageView"]);
+
+Tracking between multiple domains
+`````````````````````````````````
+To setup tracking between multiple domains you must use multiple functions ``setDomains`` to set a list of domains and
+``enableCrossDomainLinking`` to enable cross domain linking::
+
+    _paq.push(["setDomains", domains]);
+
+.. data:: domains
+
+    **Required** Domains array, with wildcards
+
+::
+
+    _paq.push(["enableCrossDomainLinking"]);
+
+Tracking subdirectories of domain in separate websites.
+```````````````````````````````````````````````````````
+To differentiate parts of website as another site for tracker user must do::
+
+    _paq.push(["setSiteId", Q]);
```

Comment:
What's `Q`? It wasn't used in previous examples.

Update it's explained bellow, but it's confusing to read in this order.

Reply:
where Q is unique site id. It was written BELOW